handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -461,7 +464,7 @@ public final long sslCtxPointer() {
     //            get access to an OpenSslSessionContext after this method was called to prevent the user from
     //            producing a segfault.
     final void destroy() {
-        synchronized (ReferenceCountedOpenSslContext.class) {
+        if (DESTROYED_UPDATER.compareAndSet(this, 0, 1)) {
```

Comment:
does this even need to be atomic? shouldn't the `AbstractReferenceCounted` guarantee this is only called a single time? It also doesn't look like we use `destroyed` or `DESTROYED_UPDATER` anywhere else.